hr_employee_service/models/hr_employee.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (C) 2018 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api
+
+from dateutil.relativedelta import relativedelta
+
+
+class HrEmployee(models.Model):
+    _inherit = 'hr.employee'
+
+    service_hire_date = fields.Date(
+        string='Hire Date',
+        groups='hr.group_hr_user',
+        track_visibility='onchange',
+        help=(
+            'Hire date is normally the date an employee completes new hire'
+            ' paperwork'
+        ),
+    )
+    service_start_date = fields.Date(
+        string='Start Date',
+        groups='hr.group_hr_user',
+        track_visibility='onchange',
+        help=(
+            'Start date is the first day the employee actually works and'
+            ' this date is used for accrual leave allocations calculation'
+        ),
+    )
+    service_termination_date = fields.Date(
+        string='Termination Date',
+        groups='hr.group_hr_user',
+        track_visibility='onchange',
+        help=(
+            'Termination date is the last day the employee actually works and'
+            ' this date is used for accrual leave allocations calculation'
+        ),
+    )
+    service_duration = fields.Float(
+        string='Service Duration',
+        digits=(2, 1),
+        readonly=True,
+        compute='_compute_service_duration',
```

Comment:
Explain to the user that this is months.

Reply:
Using `help`? And actually these are years

Reply:
OK. Using help would be fine, but using the String label would be better.

I don't want to complicate your design, but in the public sector I've seen service times represented in years, months and days (they use it for career progression). Suggestion: maybe an additional computed field with a text description of the service time.

Reply:
s/"Service Duration"/"Service Years" ?

Reply:
Better, actually I liked your idea about years/months/days

Reply:
Done! Thanks for the idea!